### PR TITLE
Add the xx_CC_COMPILER_VENDOR test

### PR DESCRIPTION
**Pull request by rhc54**
The various OAC users have near-identical versions of CC_COMPILER_VENDOR. The OMPI version appears to be more current in its checks for IBM compilers, and also has support for CXX in it - however, the CXX support isn't called anywhere and therefore is safely removed.

Checked with PMIx and OMPI - no issues found.

Signed-off-by: Ralph Castain <rhc@pmix.org>

dd

Signed-off-by: Ralph Castain <rhc@pmix.org>

### CODE REVIEW

**Comment by bwbarrett** [comment]
I'm going to squash the two commits before we merge, but wanted to do a quick review of the two steps independently.  I don't think you'll want to m4_copy this macro like others, but instead provide a helper like:

```
AC_DEFUN([PMIX_C_COMPILER_VENDOR],
[OAC_C_COMPILER_VENDOR
$1=${oac_cv_c_compiler_vendor}
AS_IF([test $1 = "clang"], [$1="gnu"],
          [test $1 = "gccfss"], [AC_MSG_ERROR([We don't support GCCFSS"])])
])
```

This was typed in the github window and not tested, but the idea is that (1) we need to handle the aborting business logic somewhere, and (2) we need to deal with the fact that all the tests in the projects mean "clang or gcc" when today they say gnu.

**Comment by rhc54** [comment]
If you are waiting for me, I'm not allowed to review my own PR - and wouldn't have the expertise to do so on this one anyway. I'm okay with committing it and then testing it against PMIx, if that helps.

**Comment by rhc54** [comment]
How important do you think it is (if at all) to update the PMIx/PRRTE release branches to use the new submodule? I'm of two minds myself - not sure of the best answer. Seems low risk enough, but really necessary right now - or maybe just easier to maintain going forward?

**Comment by bwbarrett** [comment]
I'm not planning on updating ompi before 5.0, and only updating 5.0 because it's still largely tracking master.  At this point, we haven't fixed any bugs that are critical in any of these macros from their ancestors.

I actually did all the testing for my patch on PMIx, so once we merge this (I'll squash and merge shortly), I'll post a PMIx update.

**Review by bwbarrett — APPROVED** [review]

**Pull request merged**
